src/hooks/requests/useVolumes.ts

```diff
@@ -0,0 +1,210 @@
+          getNewSubProjectWithAssignment(data, oldData)
+      )
+    },
+  })
+
+  return {
+    addAssignmentVolume,
+    isLoading,
+  }
+}
+
+export const useAssignmentEditVolume = ({
+  subProjectId: id,
+}: {
+  subProjectId?: string
+}) => {
+  const queryClient = useQueryClient()
+  const { mutateAsync: editAssignmentVolume, isLoading } = useMutation({
+    mutationKey: ['subprojects', id],
+    mutationFn: (payload: { data: ManualVolumePayload; volumeId: string }) =>
+      apiClient.put(`${endpoints.VOLUMES}/${payload.volumeId}`, payload.data),
+    onSuccess: ({ data }: { data: VolumeValue }) => {
+      queryClient.setQueryData(
+        ['subprojects', id],
+        (oldData?: SubProjectResponse) =>
+          getNewSubProjectWithAssignment(data, oldData)
+      )
+    },
+  })
+
+  return {
+    editAssignmentVolume,
+    isLoading,
+  }
+}
+
+export const useAssignmentAddCatVolume = ({
+  subProjectId: id,
+}: {
+  subProjectId?: string
+}) => {
+  const queryClient = useQueryClient()
+  const { mutateAsync: addAssignmentCatVolume, isLoading } = useMutation({
+    mutationKey: ['subprojects', id],
+    mutationFn: (payload: { data: CatVolumePayload }) =>
+      apiClient.post(`${endpoints.VOLUMES}/cat-tool`, payload.data),
+    onSuccess: ({ data }: { data: VolumeValue }) => {
+      queryClient.setQueryData(
+        ['subprojects', id],
+        (oldData?: SubProjectResponse) =>
+          getNewSubProjectWithAssignment(data, oldData)
+      )
+    },
+  })
+
+  return {
+    addAssignmentCatVolume,
+    isLoading,
+  }
+}
+
+export const useAssignmentEditCatVolume = ({
+  subProjectId: id,
+}: {
+  subProjectId?: string
+}) => {
+  const queryClient = useQueryClient()
+  const { mutateAsync: editAssignmentCatVolume, isLoading } = useMutation({
+    mutationKey: ['subprojects', id],
+    mutationFn: (payload: { data: CatVolumePayload; volumeId: string }) =>
+      apiClient.put(
+        `${endpoints.VOLUMES}/cat-tool/${payload.volumeId}`,
+        payload.data
+      ),
+    onSuccess: ({ data }: { data: VolumeValue }) => {
+      queryClient.setQueryData(
+        ['subprojects', id],
+        (oldData?: SubProjectResponse) =>
+          getNewSubProjectWithAssignment(data, oldData)
+      )
+    },
+  })
+
+  return {
+    editAssignmentCatVolume,
+    isLoading,
+  }
+}
+
+export const useAssignmentRemoveVolume = ({
+  subProjectId: id,
+}: {
+  subProjectId?: string
+}) => {
+  const queryClient = useQueryClient()
+  const { mutateAsync: removeAssignmentVolume, isLoading } = useMutation({
+    mutationKey: ['subprojects', id],
+    mutationFn: (payload: { volumeId?: string }) =>
+      apiClient.delete(`${endpoints.VOLUMES}/${payload.volumeId}`),
+    onSuccess: (_, { volumeId }) => {
+      queryClient.setQueryData(
+        ['subprojects', id],
+        (oldData?: SubProjectResponse) => {
+          const { data: previousData } = oldData || {}
+          if (!previousData) return oldData
```

Comment:
Not a blocker but maybe also extract the logic here into a separate method like getNewSubProjectWithAssignment

Reply:
Yeah not a bad idea, I usually do it once I have at least 2 places where I need to use something, but could be done before here as well